lib/configuration.rb

```diff
@@ -200,7 +200,15 @@ def setup_compiler_build_generator(compiler)
     return compiler[:build_generator] unless compiler[:build_generator].nil?
 
     if compiler[:name].match(/.*Visual Studio.*/i)
-      'Visual Studio 16 2019'
+      found_version = compiler.fetch(:version, nil)
+      case found_version
+      when 16
+        'Visual Studio 16 2019'
+      when 17
+        'Visual Studio 17 2022'
+      else
+        raise CannotMatchCompiler, 'For Visual Studio, must specify version as either 16 or 17'
+      end
```

Comment:
Instead of just assuming 2019, we need the compiler to pass in a version when doing Visual Studio builds.  This now allows Decent to support either.